openverse_catalog/dags/database/report_pending_reported_media.py

```diff
@@ -0,0 +1,171 @@
+}
+
+# Column name constants
+URL = "identifier"
+STATUS = "status"
+REASON = "reason"
+
+ReportCountsByReason = dict[str, int]
+MediaTypeReportCounts = dict[str, ReportCountsByReason]
+
+
+def get_pending_report_counts(
+    db_conn_id: str, media_type: str, ti: TaskInstance
+) -> ReportCountsByReason:
+    """
+    For the given media type, builds a dict of pending report counts grouped by
+    report reason.
+
+    Required Arguments:
+
+    db_conn_id: Connection ID for the API database
+    media_type: Media type for which to look up reports
+    ti:         Running task instance
+    """
+    postgres = PostgresHook(postgres_conn_id=db_conn_id)
+    report_counts_by_reason = {}
+
+    if media_type not in REPORTS_TABLES:
+        raise AirflowException(f"{media_type} has no Reports table configured.")
+
+    query = dedent(
+        f"""
+        SELECT
+        COUNT(DISTINCT {URL}), {REASON}
+        FROM {REPORTS_TABLES[media_type]}
+        WHERE {STATUS}='pending_review'
+        GROUP BY {REASON}
+        """
+    )
+    for count, reason in postgres.get_records(query):
+        report_counts_by_reason[reason] = count
+
+    logger.info(f"{report_counts_by_reason} records require manual review")
+
+    return report_counts_by_reason
+
+
+def report_actionable_records(
+    report_counts_by_media_type: MediaTypeReportCounts,
+) -> None:
+    """
+    Build a report detailing pending reports for all media types, and alert to Slack.
+
+    Required Arguments:
+
+    report_counts_by_media_type: dict containing report counts per media type, per
+                                 report reason
+    """
+    if all(len(reports) == 0 for reports in report_counts_by_media_type.values()):
+        slack.send_message(
+            "No records require review at this time :tada:",
+            username="Reported Media Check-In",
+        )
+        return
+
+    admin_url = os.getenv("DJANGO_ADMIN_URL", "http://localhost:8000/admin")
+    media_type_reports = ""
+
+    for media_type, distinct_report_counts in report_counts_by_media_type.items():
+        if not distinct_report_counts:
+            continue
+
+        admin_review_link = urljoin(
+            admin_url, f"api/{media_type}report/?status__exact=pending_review"
+        )
+        total_report_count = 0
+        counts_by_reason = []
+
+        for reason, count in distinct_report_counts.items():
+            total_report_count += count
+            counts_by_reason.append(f"{count} {reason}")
+
+        media_type_reports += (
+            f"  - *<{admin_review_link}|{media_type}: {total_report_count}>*"
+        )
+        media_type_reports += f" _({', '.join(counts_by_reason)})_"
+        media_type_reports += "\n"
+
+    message = f"""
+The following media have been reported and require manual review:
+{media_type_reports}
+"""
+
+    logger.info(message)
+    slack.send_alert(message, username="Reported Media Requires Review")
+
+
+def create_dag():
```

Comment:
This totally isn't necessary since it's already written, but small DAGs like this would be a perfect trial for the [TaskFlow API!](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html). I'm not sure how dynamic DAGs would work in it though... 🤔 maybe just a loop calling the `airflow.decorators.dag` decorator.